core/data/src/main/java/com/serranoie/itinero/core/data/remote/repository/TravelRepositoryImpl.kt

```
@@ -21,61 +33,148 @@ class TravelRepositoryImpl(
     }
+
+    override suspend fun updateTripInfo(tripId: String, request: UpdateTrip): Result<Trip> {
+        return safeApiCall {
+            api.updateTripInfo(tripId, request)
+            // After updating, fetch the updated trip data
+            try {
+                val updatedTrip = api.getTripById(tripId).toDomain()
+                // Update cache with fresh data
+                localRepository.cacheTrip(updatedTrip)
+                updatedTrip
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)